### PR TITLE
Implementation of group_duration

### PR DESCRIPTION
**Pull request by ragnarlonn**

### CODE REVIEW

File: js/modules/k6/k6.go [hunk 1]
```diff
@@ -59,7 +59,17 @@ func (*K6) Group(ctx context.Context, name string, fn goja.Callable) (goja.Value
 	state.Group = g
 	defer func() { state.Group = old }()
 
-	return fn(goja.Undefined())
+	tags := make(map[string]string)
+	tags["group"] = g.Path
```
Comment (liclac) [inline js/modules/k6/k6.go]: Just add the tags inline when you send out the sample. No need for a dedicated variable when there's nothing changing them.

File: js/modules/k6/k6.go [hunk 2]
```diff
@@ -59,7 +59,17 @@ func (*K6) Group(ctx context.Context, name string, fn goja.Callable) (goja.Value
...
+	// Call group function and generate group_duration sample
```
Comment (liclac) [inline js/modules/k6/k6.go]: No need for this, it's fairly obvious.

File: js/modules/k6/k6.go [hunk 3]
```diff
@@ -59,7 +59,17 @@ func (*K6) Group(ctx context.Context, name string, fn goja.Callable) (goja.Value
...
+	startTime := time.Now()
+	ret, err := fn(goja.Undefined())
+	t := time.Now()
+	duration := t.Sub(startTime).Seconds() * 1000
```
Comment (liclac) [inline js/modules/k6/k6.go]: No need for this, use `stats.D(t.Sub(startTime))`, and also do it inline when creating the sample. You can split the sample up over multiple lines.

Comment (ragnarlonn) [inline js/modules/k6/k6.go]: @liclac ok, fixed all of the above (I think)

**Pull request merged**